packages/core/src/stateUtils.ts

```diff
@@ -1518,12 +1518,30 @@ function resolveActionsAndContextWorker(
             : action.params
           : undefined;
 
+    function executeAction() {
+      actorScope.system._sendInspectionEvent({
+        type: '@xstate.action',
+        actorRef: actorScope.self,
+        action: {
+          type:
+            typeof action === 'string'
+              ? action
+              : typeof action === 'object'
+                ? action.type
+                : action.toString(),
```

Comment:
How helpful is stringifying the function? I'd be more content with a simpler version like this one: 
```suggestion
                : action.name || '(anonymous)',
```